src/sentry/search/events/builder.py

```diff
@@ -1600,12 +1603,20 @@ def __init__(
         # Don't do any of the actions that would impact performance in anyway
         # Skips all indexer checks, and won't interact with clickhouse
         self.dry_run = dry_run
+        assert dataset is None or dataset in [Dataset.PerformanceMetrics, Dataset.Metrics]
         super().__init__(
-            # Dataset is always Metrics
-            Dataset.Metrics,
+            Dataset.Metrics if dataset is None else dataset,
             *args,
             **kwargs,
         )
+        if dataset is Dataset.PerformanceMetrics:
```

Comment:
nit: python `is` keyword is used to test object identity. I think better to rewrite as:
```suggestion
        if isinstance(dataset, Dataset.PerformanceMetrics)
```

Reply:
Are you sure that applies here? Dataset is a class of Enums

Reply:
nvm yeah you are right, you want to check against `Dataset.PerformanceMetrics` which is member of Enum.